src/NLog/Config/ConfigurationItemFactory.cs

```diff
@@ -223,13 +223,70 @@ public void RegisterItemsFromAssembly(Assembly assembly, string itemNamePrefix)
         {
             InternalLogger.Debug("ScanAssembly('{0}')", assembly.FullName);
             var typesToScan = assembly.SafeGetTypes();
+            PreloadAssembly(typesToScan);
             foreach (IFactory f in this.allFactories)
             {
                 f.ScanTypes(typesToScan, itemNamePrefix);
             }
         }
 
         /// <summary>
+        /// Call Preload for NLogPackageLoader
+        /// </summary>
+        /// <param name="typesToScan"></param>
+        public void PreloadAssembly(Type[] typesToScan)
+        {
+            var types = typesToScan.Where(t => t.Name.Equals("NLogPackageLoader", StringComparison.OrdinalIgnoreCase));
+
+            foreach (var type in types)
+            {
+                CallPreload(type);
+            }
+        }
+
+        /// <summary>
+        /// Call Preload for <paramref name="type"/>. The method should be static.
```

Comment:
It would be improved as:

Calls the preload method for `<paramref name="type"/>`. The preload method should be static.